src/js.c

```diff
@@ -1933,7 +1935,7 @@ _fetch(natsMsgList *list, natsSubscription *sub, jsFetchRequest *req, bool simpl
         if ((s == NATS_OK) && sendReq)
         {
             sendReq = false;
-            req->Batch = req->Batch - (int64_t) count;
+            req->Batch = batch - (int64_t) count;
```

Comment:
Not related to your changes, but there was no need (and is actually wrong) to cast `count` to `int64_t` since it looks like both `req->Batch` and `count` are `int`, not `int64_t`. Could you fix that as part of this PR?

Reply:
Sure, fixed